internal/api/sms_provider/twilio.go

```diff
@@ -125,18 +84,17 @@ func (t *TwilioProvider) SendSms(phone string, message string) error {
 	r.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 	r.SetBasicAuth(t.Config.AccountSid, t.Config.AuthToken)
 	res, err := client.Do(r)
+	defer utilities.SafeClose(res.Body)
 	if err != nil {
 		return err
 	}
-	if res.StatusCode/100 != 2 {
+	if res.StatusCode != http.StatusOK && res.StatusCode != http.StatusCreated {
```

Comment:
there's also other 2xx status codes, which is why this was used initially
```suggestion
	if res.StatusCode/100 != 2 {
```

Reply:
Hmm this was the discussion thread  around why it was originally amended - https://github.com/supabase/gotrue/pull/981#discussion_r1116666460

 I think the other 200 status codes are not commonly used

Reply:
hmm but then this makes us reliant on the error code returned by twilio

Reply:
There are only 3 status codes that make sense: 200, 201 and 204. Let's just use what the Twilio docs say: 201. Return an error if the status code is not 201.

https://www.twilio.com/docs/sms/api/message-resource#response-status-codes